### PR TITLE
Make api respond not found

### PR DESCRIPTION
**Pull request by rafaeljesus**
## What does this PR do?
The janus API was responding 500 when no proxy found. This PR fix it.

### CODE REVIEW

File: pkg/api/mongodb_repository.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ func (r *MongoRepository) FindByName(name string) (*Definition, error) {
 	defer session.Close()
 
 	err := coll.Find(bson.M{"name": name}).One(&result)
+	if err != nil {
+		return nil, ErrAPIDefinitionNotFound
```
Comment (vgarvardt) [inline pkg/api/mongodb_repository.go]: this can be mongo connection error - is there a way to check if this is real error or result error?

**Review by vgarvardt — CHANGES_REQUESTED** [review]

File: pkg/api/mongodb_repository.go (hunk 1 above)
Comment (rafaeljesus) [reply pkg/api/mongodb_repository.go]: I will try

File: pkg/api/mongodb_repository.go [hunk 2]
```diff
@@ -51,6 +51,13 @@ func (r *MongoRepository) FindByName(name string) (*Definition, error) {
 	defer session.Close()
 
 	err := coll.Find(bson.M{"name": name}).One(&result)
+	if err != nil {
+		if err == mgo.ErrNotFound {
```
Comment (rafaeljesus) [inline pkg/api/mongodb_repository.go:55]: @vgarvardt wdyt?

Comment (rafaeljesus) [reply pkg/api/mongodb_repository.go:55]: https://github.com/go-mgo/mgo/blob/v2/session.go#L2330

**Review by vgarvardt — APPROVED** [review]

**Comment by vgarvardt** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/138/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

**Pull request merged**